File: win-prog/ProgWinCS/The Timer and Time/AnalogClock/AnalogClock.cs
//-------------------------------------------
// AnalogClock.cs © 2001 by Charles Petzold
//-------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;
using Petzold.ProgrammingWindowsWithCSharp;

class AnalogClock: Form
{
     ClockControl clkctl;

     public static void Main()
     {
          Application.Run(new AnalogClock());
     }
     public AnalogClock()
     {
          Text = "Analog Clock";
          BackColor = SystemColors.Window;
          ForeColor = SystemColors.WindowText;

          clkctl = new ClockControl();
          clkctl.Parent    = this;
          clkctl.Time      = DateTime.Now;
          clkctl.Dock      = DockStyle.Fill;
          clkctl.BackColor = Color.Black;
          clkctl.ForeColor = Color.White;

          Timer timer    = new Timer();
          timer.Interval = 100;
          timer.Tick    += new EventHandler(TimerOnTick);
          timer.Start();
     }
     void TimerOnTick(object obj, EventArgs ea)
     {
          clkctl.Time = DateTime.Now;
     }
}

File: win-prog/ProgWinVB/The Timer and Time/AnalogClock/AnalogClock.vb
'---------------------------------------------
' AnalogClock.vb (c) 2002 by Charles Petzold
'---------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class AnalogClock
    Inherits Form

    Private clkctrl As ClockControl

    Shared Sub Main()
        Application.Run(new AnalogClock())
    End Sub

    Sub New()
        Text = "Analog Clock"
        BackColor = SystemColors.Window
        ForeColor = SystemColors.WindowText

        clkctrl = New ClockControl()
        clkctrl.Parent = Me
        clkctrl.Time = DateTime.Now
        clkctrl.Dock = DockStyle.Fill
        clkctrl.BackColor = Color.Black
        clkctrl.ForeColor = Color.White

        Dim tmr As New Timer()
        tmr.Interval = 100
        AddHandler tmr.Tick, AddressOf TimerOnTick
        tmr.Start()
    End Sub

    Private Sub TimerOnTick(ByVal obj As Object, ByVal ea As EventArgs)
        clkctrl.Time = DateTime.Now
    End Sub
End Class
